src/main/integrations/electron-breadcrumbs.ts

```diff
@@ -0,0 +1,221 @@
+
+  private readonly _options: ElectronBreadcrumbsOptions<EventFunction | undefined>;
+
+  /**
+   * @param _options Integration options
+   */
+  public constructor(options: Partial<ElectronBreadcrumbsOptions<EventTypes>> = {}) {
+    this._options = {
+      unresponsive: options.unresponsive != false,
+      app: getOptionOrDefault(options, 'app'),
+      autoUpdater: getOptionOrDefault(options, 'autoUpdater'),
+      webContents: getOptionOrDefault(options, 'webContents'),
+      browserWindow: getOptionOrDefault(options, 'browserWindow'),
+      screen: getOptionOrDefault(options, 'screen'),
+      powerMonitor: getOptionOrDefault(options, 'powerMonitor'),
+    };
+  }
+
+  /** @inheritDoc */
+  public setupOnce(): void {
+    const initOptions = getCurrentHub().getClient<NodeClient>()?.getOptions() as ElectronMainOptions | undefined;
+
+    void whenAppReady.then(() => {
+      // We can't access these until app 'ready'
+      if (this._options.screen) {
+        this._instrumentBreadcrumbs('screen', screen, (name) => this._options.screen?.(name));
+      }
+
+      if (this._options.powerMonitor) {
+        this._instrumentBreadcrumbs('powerMonitor', powerMonitor, (name) => this._options.powerMonitor?.(name));
+      }
+    });
+
+    if (this._options.app) {
+      this._instrumentBreadcrumbs('app', app, (event) => this._options.app?.(event));
+    }
+
+    if (this._options.autoUpdater) {
+      this._instrumentBreadcrumbs('autoUpdater', autoUpdater, (name) => this._options.autoUpdater?.(name));
+    }
+
+    if (this._options.browserWindow) {
+      app.on('browser-window-created', (_, window) => {
+        // SetImmediate is required for contents.id to be correct in older versions of Electron
+        // https://github.com/electron/electron/issues/12036
+        setImmediate(() => {
+          if (window.isDestroyed()) {
+            return;
+          }
+          const windowName = initOptions?.getRendererName?.(window.webContents) || 'window';
+          this._instrumentBreadcrumbs(windowName, window, (name) => this._options.browserWindow?.(name));
+        });
+      });
+    }
+
+    if (this._options.webContents || this._options.unresponsive != false) {
+      app.on('web-contents-created', (_, contents) => {
+        // SetImmediate is required for contents.id to be correct in older versions of Electron
+        // https://github.com/electron/electron/issues/12036
+        setImmediate(() => {
+          if (contents.isDestroyed()) {
+            return;
+          }
+
+          const webContentsName = initOptions?.getRendererName?.(contents) || 'renderer';
+
+          if (this._options.webContents) {
+            this._instrumentBreadcrumbs(webContentsName, contents, (name) => this._options.webContents?.(name));
+          }
+
+          if (this._options.unresponsive != false) {
+            contents.on('unresponsive', () => {
+              captureMessage(`${webContentsName} Unresponsive`);
```

Comment:
should we give this a `error` (or above) Severity level?

Reply:
I've decided to pull `unresponsive` out of this integration as it doesn't really fix in here.

I'll create a new integration in another PR to address this and also other Electron events where you might want to generate Sentry rather than breadcrumbs (ie. #335)